### PR TITLE
Developers: Contribution Guidelines

### PR DESCRIPTION
**Pull request by DavidBakerEffendi**
To avoid repeating errors of first-time contributors, added a contributor guideline.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,23 +78,38 @@ docker run --rm -it -v /tmp:/tmp -v $(pwd):/app:rw -w /app -t ghcr.io/joernio/jo
...
+
+* Try make use of the templates as far as possible, however they may not suite all needs. The minimum we would like to see is:
+    - A title that briefly describes the change and purpose of the PR, preferably with the affected module in square brackets, e.g. `[javasrc2cpg] Addition Operator Fix`.
+    - A short description of the changes in the body of the PR. This could be in bullet points or paragraphs.
+    - A link or reference to the related issue, if any exists.
+* Do not:
+    - Immediately CC/@/email spam other contributors, the team will review the PR and assign the most appropriate contributor to review the PR. Joern is maintained by industry partners and researchers alike, for the most part with their own goals and priorities, and additional help is largely volunteer work. If your PR is going stale, then reach out to us in follow-up comments with @'s asking for an explanation of priority or planning of when it may be addressed (if ever, depending on quality).
+    - Leave the description body empty, this makes reviewing the purpose of the PR difficult.
+* Remember to format your code with `scalafmt`. Use `sbt scalafmt test:scalafmt`
```
Comment (mpollmeier) [inline README.md:96]:
```suggestion
* Remember to format your code, i.e. run `sbt scalafmt Test/scalafmt`
```

**Review by max-leuthaeuser — APPROVED** [review]

**Review by mpollmeier — APPROVED** [review]
💯

**Comment by DavidBakerEffendi** [comment]
We also may want to have something like https://www.scivision.dev/github-actions-selective-run/

**Comment by mpollmeier** [comment]
> We also may want to have something like https://www.scivision.dev/github-actions-selective-run/

You mean to skip CI if there's only changes in the readme, or what did you have in mind?

**Comment by DavidBakerEffendi** [comment]
@mpollmeier check this: https://github.com/joernio/joern/issues/3803 

Basically, if I make an edit to a `.scala` file in `javasrc2cpg`, then only run tests for that module. If I make edits to non-build/source code, then run no tests. If I edit `x2cpg`, then run all frontends, etc.,

**Comment by mpollmeier** [comment]
Hmm, you gotta be really careful there... e.g. javasrc2cpg is a good example where we do want to run all tests (for the entire build) if something was changed. It's used in downstream dependencies within this build, and if we don't test in the PR we could end up with a red master build.

**Comment by DavidBakerEffendi** [comment]
Right, I suppose the benign case of at least not running code when editing `.md` files may be a good minimal start

**Pull request merged**